### PR TITLE
Fix invalid robot names

### PR DESCRIPTION
**Pull request by ad-daniel**
**Description**

Fixes https://github.com/cyberbotics/webots/issues/5452

This PR:
- If the percentage-encoded robot name length is beyond an arbitrary size of 70 characters, a hash is used instead. The risk is that if the Robot name contains non-ASCII characters, the percentage encoding can result in a huge string which ends up crashing the controller when the connection takes place (QLocalServer has a limit of 106 characters for the server name)
- Reverts https://github.com/cyberbotics/webots/pull/5484 so that "/tmp" is used instead of `TMP_DIR` and changes the permissions so that multiple users can

### CODE REVIEW

**Comment by ad-daniel** [comment]
I've tested it on linux and macos, couldn't test on windows

**Review by omichel — CHANGES_REQUESTED** [review]
It looks very good to me.
I am now testing on Windows different use-cases, including local and remote controllers.

**Review by omichel — DISMISSED** [review]

**Comment by omichel** [comment]
I would also add a line in the change log.

**Review by omichel — APPROVED** [review]
Thank you.

File: src/webots/control/WbController.cpp [hunk 1]
```diff
@@ -314,6 +321,9 @@ void WbController::start() {
   }
   file.write("");
   file.close();
+
+  // FIXME: from Qt 6.4 onwards, QDir::mkdir can be used to set the permissions
+  QProcess::execute("sh", QStringList() << "-c" << QString("chmod -R 777 %1").arg(mIpcPath));
```
Comment (omichel) [inline src/webots/control/WbController.cpp:326]: This code seems wrong: useless and not working on Windows where it is supposed to be executed.

**Pull request merged**